cli/cmd/edges.go

```diff
@@ -0,0 +1,297 @@
+			client := checkPublicAPIClientOrExit()
+			c := make(chan indexedEdgeResults, len(reqs))
+			for num, req := range reqs {
+				go func(num int, req *pb.EdgesRequest) {
+					resp, err := requestEdgesFromAPI(client, req)
+					if err != nil {
+						fmt.Println(err)
+					}
+
+					rows := edgesRespToRows(resp)
+
+					c <- indexedEdgeResults{num, rows, err}
+				}(num, req)
+			}
+
+			totalRows := make([]*pb.Edge, 0)
+			i := 0
+			for res := range c {
+				if res.err != nil {
+					return res.err
+				}
+				totalRows = append(totalRows, res.rows...)
+				if i++; i == len(reqs) {
+					close(c)
+				}
+			}
+
+			output := renderEdgeStats(totalRows, options)
+			_, err = fmt.Print(output)
+
+			return err
+		},
+	}
+
+	cmd.PersistentFlags().StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace of the specified resource")
+	cmd.PersistentFlags().StringVarP(&options.outputFormat, "output", "o", options.outputFormat, "Output format; one of: \"table\" or \"json\"")
+	return cmd
+}
+
+func buildEdgesRequests(resources []string, options *statOptions) ([]*pb.EdgesRequest, error) {
+	targets, err := util.BuildResources(options.namespace, resources)
+	if err != nil {
+		return nil, err
+	}
+
+	requests := make([]*pb.EdgesRequest, 0)
+	for _, target := range targets {
+		err = options.validate(target.Type)
+		if err != nil {
+			return nil, err
+		}
+
+		requestParams := util.EdgesRequestParams{
+			ResourceType: target.Type,
+			Namespace:    options.namespace,
+		}
+
+		req, err := util.BuildEdgesRequest(requestParams)
+		if err != nil {
+			return nil, err
+		}
+		requests = append(requests, req)
+	}
+	return requests, nil
+}
+
+func edgesRespToRows(resp *pb.EdgesResponse) []*pb.Edge {
+	rows := make([]*pb.Edge, 0)
+	if resp != nil {
+		for _, edgeTable := range resp.GetOk().Edges {
+			rows = append(rows, edgeTable)
+		}
+	}
+	return rows
+}
+
+func requestEdgesFromAPI(client pb.ApiClient, req *pb.EdgesRequest) (*pb.EdgesResponse, error) {
+	resp, err := client.Edges(context.Background(), req)
+	if err != nil {
+		return nil, fmt.Errorf("Edges API error: %+v", err)
+	}
+	if e := resp.GetError(); e != nil {
+		return nil, fmt.Errorf("Edges API response error: %+v", e.Error)
+	}
+	return resp, nil
+}
+
+func renderEdgeStats(rows []*pb.Edge, options *statOptions) string {
+	var buffer bytes.Buffer
+	w := tabwriter.NewWriter(&buffer, 0, 0, padding, ' ', tabwriter.AlignRight)
+	writeEdgesToBuffer(rows, w, options)
+	w.Flush()
+
+	return renderEdges(buffer, &options.statOptionsBase)
+}
+
+type edgeRowStats struct {
+	src    string
+	dst    string
+	client string
+	server string
+	msg    string
+}
+
+func writeEdgesToBuffer(rows []*pb.Edge, w *tabwriter.Writer, options *statOptions) {
+	edgeTables := make(map[string]*edgeRowStats)
+	if len(rows) != 0 {
+		for _, r := range rows {
+			key := string(r.Dst.Name + r.Src.Name)
+
+			edgeTables[key] = &edgeRowStats{
+				client: r.ClientId,
+				server: r.ServerId,
```

Comment:
are the identities consistently of the form `resource.namespace.serviceaccount.identity.linkerd.cluster.local`. if so, can we drop the `.serviceaccount.identity.linkerd.cluster.local`, it makes the table output wrap on my laptop screen:
<img width="1273" alt="Screen Shot 2019-05-10 at 12 39 44 PM" src="https://user-images.githubusercontent.com/236915/57521490-b7b8cc80-7320-11e9-81f4-592789e387d6.png">

Reply:
Yup, that's the consistent format of the identities.